app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,17 @@
+class VotesController < ApplicationController
+  before_action :set_voteable, only: [:upvote, :downvote]
+  
+  def upvote
+    @voteable.upvote(current_user.id)
+    render json: { vote_count: @voteable.total_upvotes - @voteable.total_downvotes, type: 'upvote' }
```

Comment:
@voteable.total_upvotes - @voteable.total_downvotes should be a method from the module